[Bug] Crash when pressing Shift-W in rules tile selector

Repro steps:
1. Open rules editor
2. Add new rule
3. Click the tile selector to open a view of the tilemap
4. Press Shift-W

Video:
https://user-images.githubusercontent.com/5410234/213885177-7bddfa23-0aaa-424e-afd0-c2dcb662dc64.mov


Stack:
```
LDtk version: 1.2.5-
Empty jQuery object
Error

```
Processes:
```
#0 App
 |--- #2050 page.Editor
       |--- #2051 misc.FileWatcher
       |--- #2052 display.WorldRender
       |--- #2053 display.LevelRender [PAUSED]
       |--- #2054 display.Camera
       |--- #2055 display.Rulers[--]
       |--- #2056 tool.SelectionTool[--]
       |--- #2057 tool.lt.DoNothing[--]
       |--- #2058 WorldTool
       |--- #2059 tool.PanView[--]
       |--- #2392 tool.lt.IntGridTool[--]
       |--- #2467 ui.modal.panel.EditAllAutoLayerRules
       |--- #2871 ui.modal.dialog.RuleEditor
       |--- #2872 ui.Modal
       |--- #2873 ui.modal.Progress
```

Log:
```
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[GENERAL]      Added rule
[EVENT]        LayerRuleAdded 33
[FILE]         Reading HTML template ruleEditor from cache
[RENDER]       Rendering world bg...
[EVENT]        LayerRuleChanged 33
[EVENT]        LayerRuleChanged 33
[RENDER]       Rendering world bg...
[EVENT]        ToolValueSelected
[EVENT]        ToolValueSelected
[EVENT]        ToolValueSelected
[EVENT]        ToolValueSelected
[EVENT]        ToolValueSelected
[EVENT]        ToolValueSelected
[EVENT]        ToolValueSelected
[EVENT]        ToolValueSelected
[EVENT]        LayerRuleChanged 33
[RENDER]       Rendering world bg...
[EVENT]        ToolValueSelected
[EVENT]        LayerRuleChanged 33
[EVENT]        LayerRuleChanged 33
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[FILE]         Reading HTML template ruleEditor from cache
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[USERACTION]   Closed panel #2467 ui.modal.panel.EditAllAutoLayerRules
[GENERAL]      Applying invalidated rules...
[GENERAL]      Progress created.
[EVENT]        LayerRuleChanged 33
[GENERAL]      Page started: page.CrashReport()
[FILE]         Loading page template: crashReport from /Applications/LDtk.app/Contents/Resources/app.asar/assets/tpl/pages/crashReport.html
[ERROR]        Empty jQuery object (Error)
[ERROR]        undefined

[GENERAL]      
#0 App
 |--- #2050 page.Editor
       |--- #2051 misc.FileWatcher
       |--- #2052 display.WorldRender
       |--- #2053 display.LevelRender [PAUSED]
       |--- #2054 display.Camera
       |--- #2055 display.Rulers[--]
       |--- #2056 tool.SelectionTool[--]
       |--- #2057 tool.lt.DoNothing[--]
       |--- #2058 WorldTool
       |--- #2059 tool.PanView[--]
       |--- #2392 tool.lt.IntGridTool[--]
       |--- #2467 ui.modal.panel.EditAllAutoLayerRules
       |--- #2871 ui.modal.dialog.RuleEditor
       |--- #2872 ui.Modal
       |--- #2873 ui.modal.Progress
 |--- #2874 page.CrashReport

[FILE]         Cleared all file watches
```

--- a/src/electron.renderer/ui/Modal.hx
+++ b/src/electron.renderer/ui/Modal.hx
@@ -127,9 +127,11 @@ class Modal extends dn.Process {
 
 	public static function closeAll(?except:Modal) {
 		var any = false;
-		for(w in ALL)
-			if( !w.isClosing() && ( except==null || w!=except ) && w.canBeClosedManually ) {
-				w.close();
+
+		var w = ALL.length;
+		while (--w >= 0)
+			if ( !ALL[w].isClosing() && ( except==null || ALL[w]!=except ) && ALL[w].canBeClosedManually ) {
+				ALL[w].close();
 				any = true;
 			}
 		return any;

